p2p/conn/connection.go

```diff
@@ -782,3 +780,25 @@ type PacketMsg struct {
 func (mp PacketMsg) String() string {
 	return fmt.Sprintf("PacketMsg{%X:%X T:%X}", mp.ChannelID, mp.Bytes, mp.EOF)
 }
+
+// - Uvarint length of MustMarshalBinary(packet) = 1 or 2 bytes
+//   (as long as it's less than 16,384 bytes)
+// - Prefix bytes = 4 bytes
+// - ChannelID field key + byte = 2 bytes
+// - EOF field key + byte = 2 bytes
+// - Bytes field key = 1 bytes
+// - Uvarint length of MustMarshalBinary(bytes) = 1 or 2 bytes
+// - Struct terminator = 1 byte
+// = up to 14 bytes overhead for the packet.
+
+func emptyPacketMsgSize() int {
+	emptyPacketMsgSize := len(cdc.MustMarshalBinary(PacketMsg{
+		ChannelID: 0x01,
+		EOF:       1,
+		Bytes:     make([]byte, 1),
+	}))
+	// -1 byte of data
+	// +1 byte because uvarint length of MustMarshalBinary(bytes) will be 2 bytes for big packets
+	// +1 byte because uvarint length of MustMarshalBinary(packet) will be 2 bytes for big packets
+	return emptyPacketMsgSize - 1 + 1 + 1
```

Comment:
The whole thing about (u)varints is they are variable sized. In general, it might be a good idea to expose fixed-encoding sizes but for this particular example it doesn't make much sense.

Reply:
For speed we want a custom encoder so please bear with us on p2p until we make Amino compilers.